src/utils.py

```diff
@@ -219,53 +221,296 @@ def create_state_from_dict(data: Dict[Object, Dict[str, float]],
     return State(state_dict, simulator_state)
 
 
-def intersects(p1: Tuple[float, float], p2: Tuple[float, float],
-               p3: Tuple[float, float], p4: Tuple[float, float]) -> bool:
-    """Checks if line segment p1p2 and p3p4 intersect.
+class _Geom2DBody(abc.ABC):
+    """A 2D body that contains some points."""
```

Comment:
is 2D going to be enough for us, thinking ahead?

Reply:
I think so? for 3D things, I imagine we'll use PyBullet